### PR TITLE
Add pw.mmk.OpenFreebuds

### PR DESCRIPTION
**Pull request by melianmiko**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.

FOSS client for HUAWEI FreeBuds wireless earphone series. Allows to view device state and information, change built-in device settings, control active noise cancellation and etc.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/melianmiko/OpenFreebuds, https://mmk.pw/en/openfreebuds

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by melianmiko** [comment]
bot, build pw.mmk.OpenFreebuds

File: pw.mmk.OpenFreebuds.desktop
Comment (hfiguiere) [inline pw.mmk.OpenFreebuds.desktop:1]: this file should be upstream

File: pw.mmk.OpenFreebuds.metainfo.xml
Comment (hfiguiere) [inline pw.mmk.OpenFreebuds.metainfo.xml:1]: this file should be upstream

File: pw.mmk.OpenFreebuds.png
Comment (hfiguiere) [inline pw.mmk.OpenFreebuds.png:1]: the icon should be upstream

File: pw.mmk.OpenFreebuds.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    build-commands:
+      - pip3 install --prefix=/app --no-deps *.whl
+    sources:
+      - type: file
+        url: https://files.pythonhosted.org/packages/df/b3/072c28eace372ba7630ea187b7efd7f09cc8bcebf847a96b5e03e9cc0828/expandvars-0.12.0-py3-none-any.whl
+        sha256: 7432c1c2ae50c671a8146583177d60020dd210ada7d940e52af91f1f84f753b2
+
+  - python3-requirements.json
+
+  - name: PyQtApp
```
Comment (hfiguiere) [inline pw.mmk.OpenFreebuds.yml:39]:
```suggestion
  - name: openfreebuds
```

**Comment by melianmiko** [comment]
bot, build pw.mmk.OpenFreebuds

File: pw.mmk.OpenFreebuds.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+sdk: org.kde.Sdk
+runtime: org.kde.Platform
+base: com.riverbankcomputing.PyQt.BaseApp
+runtime-version: '6.7'
+base-version: '6.7'
+
+finish-args:
+  # Connect to Bluetooth devices through RFCOMM / SPP protocol
+  - --allow=bluetooth
+  - --share=network
```
Comment (bbhtt) [inline pw.mmk.OpenFreebuds.yml:13]: why is network required ? `allow=bluetooth` should be sufficient I think ?

Comment (melianmiko) [reply pw.mmk.OpenFreebuds.yml:13]: > `allow=bluetooth` should be sufficient I think ?

But it wasn't... Idk why, but Python didn't want establish a Bluetooth socket connection without network permission. Release build didn't have enough logging level to show full exception, but as I remember it reports something like `ConnectionResetError`.

Screenshot_20240928_211832

When I enable network access back, it works:

Screenshot_20240928_212116

Comment (bbhtt) [reply pw.mmk.OpenFreebuds.yml:13]: what library are you using ?

Comment (melianmiko) [reply pw.mmk.OpenFreebuds.yml:13]: For Bluetooth, nothing, Python's built-in `socket` module supports establishing connections through Bluetooth RFCOMM / SPP.

Comment (melianmiko) [reply pw.mmk.OpenFreebuds.yml:13]: Built a variant with extended logging to get an exception, and here it is, it literally can't connect without network access...

Screenshot_20240928_212815

Comment (hfiguiere) [reply pw.mmk.OpenFreebuds.yml:13]: socket access need network permission even AF_UNIX sockets. It's quite unfortunate.

Comment (melianmiko) [reply pw.mmk.OpenFreebuds.yml:13]: Looks like it's so....

Comment (bbhtt) [reply pw.mmk.OpenFreebuds.yml:13]: Yea

**Pull request closed without merging**